src/species_advance/standard/pipeline/hydro_p_pipeline.cc

```diff
@@ -0,0 +1,191 @@
+#define IN_spa
+
+#define HAS_V4_PIPELINE
+#define HAS_V8_PIPELINE
+#define HAS_V16_PIPELINE
+
+#include "spa_private.h"
+
+#include "../../../util/pipelines/pipelines_exec.h"
+
+// accumulate_hydro_p adds the hydrodynamic fields associated with the
+// supplied particle_list to the hydro array.  Trilinear interpolation
+// is used.  hydro is known at the nodes at the same time as particle
+// positions. No effort is made to fix up edges of the computational
+// domain.  All particles on the list must be inbounds.  Note, the
+// hydro jx,jy,jz are for diagnostic purposes only; they are not
+// accumulated with a charge conserving algorithm.
+
+void
+hydro_p_pipeline_scalar( hydro_p_pipeline_args_t * args,
+                         int pipeline_rank,
+                         int n_pipeline )
+{
+  float c, qsp, mspc, qdt_2mc, qdt_4mc2, r8V;
+  int np, stride_10, stride_21, stride_43;
+
+  float dx, dy, dz, ux, uy, uz, w, vx, vy, vz, ke_mc;
+  float w0, w1, w2, w3, w4, w5, w6, w7, t;
+  int i, n, n1, n0;
+
+  const species_t      *              sp = args->sp;
+  const grid_t         *              g  = sp->g;
+  /**/  hydro_t        * ALIGNED(128) h  = args->h + pipeline_rank*args->h_size;
+  const particle_t     * ALIGNED(128) p  = sp->p;
+  const interpolator_t * ALIGNED(128) f  = args->f;
+  const bool               charge_weight = args->charge_weight;
+
+  // Determine which particles this pipeline processes.
+  DISTRIBUTE( args->np, 16, pipeline_rank, n_pipeline, n0, n1 ); n1 += n0;
+
+  c        = sp->g->cvac;
+  if(charge_weight) {
+    qsp      = sp->q;
+  } else {
+    qsp      = sp->q;
```

Comment:
I'm guessing one of these should be `w`?

Reply:
Indeed. Good catch